db-engine/src/main/java/com/splicemachine/db/impl/services/reflect/UpdateLoader.java

```diff
@@ -211,6 +205,7 @@ public Object run(){
 	Class loadClass(String className, boolean resolve) 
 		throws ClassNotFoundException {
 
+		boolean isRS = Thread.currentThread().getName().contains("DRDAConnThread");
```

Comment:
If running in a NLJ, union operation or other contexts they could be running in a different thread, we need a different way to identify RS vs Olap/Spark

Reply:
Is it possible nested loop join may try loading jars?